WordPress/gutenberg
# ZStack: Convert component story to TypeScript and add inline docs

## What?
Converts the `ZStack` component story to TypeScript and update the code to follow coding standards.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactor story to TypeScript
- Move types to `types.ts`
- Update readme docs
- Added inline code example to component

## Testing Instructions
- Run npm run dev and ensure no typing errors
- Fire up the Storybook and confirm the `ZStack` continues to function as expected
